controllers/ProposalController.php

```diff
@@ -34,44 +35,13 @@ class ProposalController extends MainController
      */
     public function actionMyProposals(int $id = null): string
     {
+
         if (!is_null($id)) {
             $selectedProposal = $this->checkIfProposalExists($id);
             $this->checkIfUserIsOwnerOfProposal($selectedProposal->submitter->id);
-            $chronologicalStream = $this->generateChronologicalStream
-            (
-                $selectedProposal->comments,
-                $selectedProposal->reviews,
-                $selectedProposal->proposalContentHistories,
-                $selectedProposal->proposalFileHistories,
-
-            );
-            $lastProposalContent = $selectedProposal->proposalContentHistories[
-                count($selectedProposal->proposalContentHistories)-1
-            ];
-
-            $approvalsCount = Review::find()->where(['proposal_id' => $selectedProposal->id])
-                ->andWhere(['status' => \app\models\Review::REVIEW_STATUS_APPROVED])
-                ->count();
-
-            $disapprovalsCount = Review::find()->where(['proposal_id' => $selectedProposal->id])
-                ->andWhere(['status' => \app\models\Review::REVIEW_STATUS_DISAPPROVED])
-                ->count();
-
-            $manageProposalFormModel = new ManageProposalForm();
-            $manageProposalFormModel->title = $selectedProposal->title;
-            $manageProposalFormModel->content = $lastProposalContent->content;
-
-            $manageCommentFormModel = new ManageCommentForm();
+            $displayItems = $this->buildOneProposalDisplayItems($selectedProposal);
```

Comment:
Simple suggestion, I prefer $viewItems as variable name, because it is all the values you pass to the view.

Reply:
Deal !